pkg/util/http.go

```diff
@@ -172,14 +173,18 @@ func ParseProtoReader(ctx context.Context, reader io.Reader, expectedSize, maxSi
 	return body, nil
 }
 
+func NewMsgSizeTooLargeErr(actual, limit int) error {
+	return errors.New(globalerror.MsgSizeTooLarge.MessageWithGlobalLimitConfig(fmt.Sprintf("the incoming push request has been rejected because its message size of %d bytes is larger than the allowed limit of %d bytes", actual, limit), "distributor.max-recv-msg-size"))
```

Comment:
This looks like a generic function used to generate the error, but then it contains "distributor.max-recv-msg-size" which is very specific. This error is returned from the exported function `ParseProtoReader()` which is not just called by the remote write.

I would suggest a different approach:
1. From this package, return a globally defined static error, like `var ErrMessageTooLarge = errors.New("the request has been rejected because its message size exceed the limit")`. The message is **intentionally** generic and **static**
2. In `pkg/util/push/` check the error returned by `ParseProtoReader` and if it's a `ErrMessageTooLarge` then remap it into a more specific error (mentioning the distributor flag in this case, which is correct)

Reply:
1. I replaced it with a struct because the existing error messages are already dynamic with the actual and limit numbers included in the msg, is that okay?